projects/templedao/index.js

```diff
@@ -1,32 +1,39 @@
 const { staking } = require("../helper/staking");
-const { sumTokensAndLPsSharedOwners } = require("../helper/unwrapLPs");
+const { gql, request } = require("graphql-request");
+const { toUSDTBalances } = require("../helper/balances");
 
 const templeStakingContract = "0x4D14b24EDb751221B3Ff08BBB8bd91D4b1c8bc77";
 const TEMPLE = "0x470ebf5f030ed85fc1ed4c2d36b9dd02e77cf1b7";
 
-const templeTreasuryContract = "0x22c2fE05f55F81Bf32310acD9a7C51c4d7b4e443";
-const FRAX = "0x853d955acef822db058eb8505911ed77f175b99e";
-
-async function treasuryTvl() {
-  const balances = {};
-
-  await sumTokensAndLPsSharedOwners(
-    balances,
-    [
-      [FRAX, false],
-    ],
-    [templeTreasuryContract]
-  );
+const protocolQuery = gql`
+query get_tvl($block: Int) {
+  protocolMetrics(
+    first: 1, orderBy: timestamp, orderDirection: desc
+  ) {
+    lockedStables
+    timestamp
+  }
+}
+`
 
-  return balances;
+/*** TVL Portion (Treasury) ***
+ * Treasury TVL consists of stables from TEMPLE-FRAX AMM LP, Temple Defend and FRAX+3CRV LP
+ ***/
+async function ethTvl(timestamp, block) {
+  const queriedData = await request("https://api.thegraph.com/subgraphs/name/templedao/templedao-metrics", protocolQuery, {block})
```

Comment:
Not sure, why we need to switch to graph here. I see that old treasury contract does not have the LPs you mentioned, can you give me the (treasury) address, I could add them :)

Reply:
Hey, thanks for the quick feedback!
Using the subgraph is just a convenient way to combine multiple sources. That way we don't have to update the adapter every time we move the treasury funds.

If you still want to use the separate wallets instead, here are the addresses:
TEMPLE-FRAX AMM LP: `0x6021444f1706f15465bee85463bcc7d7cc17fc03`
Temple Defend:  `0x8a5058100e60e8f7c42305eb505b12785bba3bca + 0xb0d978c8be39c119922b99f483cd8c4092f0ea56`
FRAX+3CRV LP: `0x5c8898f8e0f9468d4a677887bc03ee2659321012`

Please let me know if you need anything else.